src/store/account/sagas.js

```diff
@@ -86,6 +92,38 @@ function* onFetchAccount(account) {
   // Add playlists that might not have made it into the user's library.
   // This could happen if the user creates a new playlist and then leaves their session.
   yield fork(addPlaylistsNotInLibrary)
+
+  yield fork(initAudioChecks)
+}
+
+/**
+ * Opens the Modal for user confirmation of transfering audio to waudio
+ */
+function* requestTransferAudioToWAudio() {
+  // Wait for any other modals to close
+  let modalIsOpen = yield select(getModalIsOpen)
+  while (modalIsOpen) {
+    yield take(setVisibility.type)
+    modalIsOpen = yield select(getModalIsOpen)
+  }
+  // Put an action to show modal
+  yield put(setVisibility({ modal: 'ConfirmAudioToWAudio', visible: true }))
+
+  // Take the action on result of modal
+  yield take(confirmTransferAudioToWAudio.type)
+  return true
+}
+
+function* initAudioChecks() {
+  if (getFeatureEnabled(FeatureFlags.TRANSFER_AUDIO_TO_WAUDIO_ON_LOAD)) {
+    const audiusManager = new AudioManager({
+      requestTransferAudioToWAudio
+    })
+
+    yield call(audiusManager.getInitState)
+    yield call(audiusManager.updateState)
+    yield put(setVisibility({ modal: 'ConfirmAudioToWAudio', visible: false }))
```

Comment:
should we try catch this and set visibility to false even in the case it fails?

Reply:
yup, moved